### PR TITLE
[Fix](Status) Handle returned Status correctly

### PR DESCRIPTION
**Pull request by wyxxxcat**
## Proposed changes

Issue Number: #28160

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wyxxxcat** [comment]
run  buildall

File: be/src/index-tools/index_tool.cpp [hunk 1]
```diff
@@ -265,7 +265,7 @@ int main(int argc, char** argv) {
                 std::vector<FileInfo> files;
                 bool exists = false;
                 std::filesystem::path root_dir(FLAGS_directory);
-                static_cast<void>(fs->list(root_dir, true, &files, &exists));
+                THROW_IF_ERROR(fs->list(root_dir, true, &files, &exists));
```
Comment (zclllyybb) [inline be/src/index-tools/index_tool.cpp:268]: There's no try-catch be able to deal with it. please print errlog and exit

File: be/src/vec/exec/vanalytic_eval_node.cpp [hunk 2]
```diff
@@ -203,15 +203,17 @@ Status VAnalyticEvalNode::prepare(RuntimeState* state) {
     }
     _fn_place_ptr = _agg_arena_pool->aligned_alloc(_total_size_of_aggregate_states,
                                                    _align_aggregate_states);
-    RETURN_IF_CATCH_EXCEPTION(static_cast<void>(_create_agg_status()));
-    _executor.insert_result =
-            std::bind<void>(&VAnalyticEvalNode::_insert_result_info, this, std::placeholders::_1);
-    _executor.execute =
-            std::bind<void>(&VAnalyticEvalNode::_execute_for_win_func, this, std::placeholders::_1,
-                            std::placeholders::_2, std::placeholders::_3, std::placeholders::_4);
+    RETURN_IF_CATCH_EXCEPTION(THROW_IF_ERROR(_create_agg_status()));
```
Comment (zclllyybb) [inline be/src/vec/exec/vanalytic_eval_node.cpp:206]: why not just `RETURN_IF_ERROR`?

File: be/src/vec/exec/vanalytic_eval_node.cpp [hunk 3]
```diff
@@ -610,7 +612,7 @@ bool VAnalyticEvalNode::_init_next_partition(BlockRowPos found_partition_end) {
         _partition_by_start = _partition_by_end;
         _partition_by_end = found_partition_end;
         _current_row_position = _partition_by_start.pos;
-        static_cast<void>(_reset_agg_status());
+        THROW_IF_ERROR(_reset_agg_status());
```
Comment (zclllyybb) [inline be/src/vec/exec/vanalytic_eval_node.cpp:615]: This function only returns OK, isn't it?

File: be/src/vec/exec/vanalytic_eval_node.cpp [hunk 4]
```diff
@@ -203,15 +203,17 @@ Status VAnalyticEvalNode::prepare(RuntimeState* state) {
...
     _fn_place_ptr = _agg_arena_pool->aligned_alloc(_total_size_of_aggregate_states,
                                                    _align_aggregate_states);
-    RETURN_IF_CATCH_EXCEPTION(static_cast<void>(_create_agg_status()));
-    _executor.insert_result =
-            std::bind<void>(&VAnalyticEvalNode::_insert_result_info, this, std::placeholders::_1);
-    _executor.execute =
-            std::bind<void>(&VAnalyticEvalNode::_execute_for_win_func, this, std::placeholders::_1,
-                            std::placeholders::_2, std::placeholders::_3, std::placeholders::_4);
+    RETURN_IF_CATCH_EXCEPTION(THROW_IF_ERROR(_create_agg_status()));
+    _executor.insert_result = [this](auto&& PH1) {
```
Comment (zclllyybb) [inline be/src/vec/exec/vanalytic_eval_node.cpp:207]: for bind to lambda, could you use the concrete type, but not `auto` for parameters?

File: be/src/vec/functions/function_totype.h [hunk 5]
```diff
@@ -18,6 +18,7 @@
 #pragma once
 #include <fmt/format.h>
 
+#include "common/status.h"
```
Comment (zclllyybb) [inline be/src/vec/functions/function_totype.h:21]: for what?

File: be/src/vec/sink/writer/vfile_result_writer.cpp [hunk 6]
```diff
@@ -306,7 +306,7 @@ Status VFileResultWriter::_send_result() {
     row_buffer.push_bigint(_written_rows_counter->value()); // total rows
     row_buffer.push_bigint(_written_data_bytes->value());   // file size
     std::string file_url;
-    static_cast<void>(_get_file_url(&file_url));
+    RETURN_IF_ERROR(_get_file_url(&file_url));
```
Comment (zclllyybb) [inline be/src/vec/sink/writer/vfile_result_writer.cpp:309]: only OK

**Review by zclllyybb — CHANGES_REQUESTED** [review]

**Comment by wyxxxcat** [comment]
run buildall

File: be/src/vec/exec/vanalytic_eval_node.cpp [hunk 7]
```diff
@@ -203,15 +203,15 @@ Status VAnalyticEvalNode::prepare(RuntimeState* state) {
     }
     _fn_place_ptr = _agg_arena_pool->aligned_alloc(_total_size_of_aggregate_states,
                                                    _align_aggregate_states);
-    RETURN_IF_CATCH_EXCEPTION(static_cast<void>(_create_agg_status()));
+    RETURN_IF_CATCH_EXCEPTION(RETURN_IF_ERROR(_create_agg_status()));
```
Comment (zclllyybb) [inline be/src/vec/exec/vanalytic_eval_node.cpp:206]: wrong

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run p0 10

**Comment by wyxxxcat** [comment]
run buildall

**Comment by wyxxxcat** [comment]
run p0 10

**Review by zclllyybb — APPROVED** [review]
LGTM

File: be/src/util/thrift_server.cpp [hunk 8]
```diff
@@ -375,7 +375,7 @@ Status ThriftServer::start() {
             new ThriftServer::ThriftServerEventProcessor(this));
     _server->setServerEventHandler(event_processor);
 
-    RETURN_IF_ERROR(event_processor->start_and_wait_for_server());
+    static_cast<void>(event_processor->start_and_wait_for_server());
```
Comment (zclllyybb) [inline be/src/util/thrift_server.cpp:378]: What's wrong before?

Comment (zclllyybb) [reply be/src/util/thrift_server.cpp:378]: We shouldn't change this. please explain this logic

**Review by zclllyybb — CHANGES_REQUESTED** [review]

**Comment by wyxxxcat** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**